src/crossChain/AxelarProvider.ts

```diff
@@ -0,0 +1,328 @@
+  }
+
+  getSupportedChains() {
+    return this.squid.chains
+  }
+
+  async executeRoute(
+    route: RouteResponse,
+    provider: Provider
+  ): Promise<ethers.providers.TransactionReceipt> {
+    const signer = await new ethers.providers.Web3Provider(provider).getSigner()
+
+    // @ts-ignore
+    // tslint:disable-next-line
+    const txResponse = (await this.squid.executeRoute({
+      route: route.route,
+      signer
+    })) as ethers.providers.TransactionResponse
+
+    return txResponse.wait()
+  }
+
+  async buyNFT(
+    provider: Provider,
+    buyNFTCrossChainData: BuyNFTCrossChainData
+  ): Promise<string> {
+    const route = await this.getBuyNFTRoute(buyNFTCrossChainData)
+    const tx = await this.executeRoute(route, provider)
+    return tx.transactionHash
+  }
+
+  async getBuyNFTRoute(
+    buyNFTCrossChainData: BuyNFTCrossChainData
+  ): Promise<RouteResponse> {
+    await this.init()
+    const {
+      fromAddress,
+      fromAmount,
+      fromChain,
+      fromToken,
+      toChain,
+      toAmount, // the item price
+      enableExpress = true, // TODO: check if we need this
+      slippage = 1, // 1 is "normal" slippage. Always set to 1
+      nft: { collectionAddress, price, tokenId }
+    } = buyNFTCrossChainData
+
+    const ERC20ContractInterface = new ethers.utils.Interface(ERC20)
+    const marketplaceContractABI =
+      toChain === ChainId.MATIC_MAINNET ? MarketplaceV2 : Marketplace
+
+    const marketplaceInterface = new ethers.utils.Interface(
+      marketplaceContractABI
+    )
+    const ERC721ContractInterface = new ethers.utils.Interface(ERC721)
+
+    const destinyChainMANA = getContract(ContractName.MANAToken, toChain)
+      .address
+
+    const destinyChainMarketplace = getContract(
+      toChain === ChainId.MATIC_MAINNET
+        ? ContractName.MarketplaceV2
+        : ContractName.Marketplace,
+      toChain
+    ).address
+
+    return this.squid.getRoute({
+      fromAddress,
+      fromAmount,
+      fromToken,
+      fromChain: fromChain.toString(),
+      toToken: destinyChainMANA,
+      toChain: toChain.toString(),
+      toAddress: destinyChainMarketplace,
+      enableBoost: enableExpress,
+      slippageConfig: {
+        autoMode: slippage
+      },
+      postHook: {
+        chainType: ChainType.EVM,
+        fundAmount: '1',
+        fundToken: destinyChainMANA,
+        calls: [
+          // ===================================
+          // Approve MANA to be spent by Decentraland contract
+          // ===================================
+          {
+            chainType: ChainType.EVM,
+            callType: SquidCallType.DEFAULT,
+            target: destinyChainMANA,
+            value: '0',
+            callData: ERC20ContractInterface.encodeFunctionData('approve', [
+              destinyChainMarketplace,
+              toAmount
+            ]),
+            payload: {
+              tokenAddress: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
```

Comment:
wdyt about moving this address to a const/another file?

Reply:
good idea, done!